plugins/pulp_docker/plugins/distributors/publish_steps.py

```diff
@@ -220,3 +223,154 @@ def process_main(self):
+                                                      repo=repo, config=config,
+                                                      remote_repo_path=remote_repo_path,
+                                                      published_unit_path=['manifests'])
+        self._tag_names = set()
+        self.repo_registry_id = repo_registry_id
+
+    def process_main(self, item=None):
+        """
+        Create the manifest tag relative links.
+
+        :param item: The tag to process
+        :type  item: pulp_docker.plugins.models.Tag
+        """
+        manifest = models.Manifest.objects.get(digest=item.manifest_digest)
+        filename = item.name
+        symlink = self.make_link_unit(manifest, filename, self.get_working_dir(),
+                                      self.remote_repo_path,
+                                      self.get_config().get("remote")["root"],
+                                      self.published_unit_path)
+        self.parent.symlink_list.append(symlink)
+        self._tag_names.add(item.name)
+
+    def finalize(self):
+        """
+        Write the Tag list file so that clients can retrieve the list of available Tags.
+        """
+        tags_path = os.path.join(self.parent.get_working_dir(), '.relative', 'tags')
+        misc.mkdir(tags_path)
+        with open(os.path.join(tags_path, 'list'), 'w') as list_file:
+            tag_data = {
+                'name': self.repo_registry_id,
+                'tags': list(self._tag_names)}
+            list_file.write(json.dumps(tag_data))
+        # We don't need the tag names anymore
+        del self._tag_names
+
+
+class DockerRsyncPublisher(Publisher):
+
+    REPO_CONTENT_TYPES = [constants.IMAGE_TYPE_ID, constants.BLOB_TYPE_ID,
+                          constants.MANIFEST_TYPE_ID]
+
+    REPO_CONTENT_MODELS = [models.Blob, models.Manifest, models.Image]
+
+    def _get_predistributor(self):
+        """
+        Returns the distributor object representing the predistributor. A predistributor is the
+        distributor that is used to publish a repository prior to publishing with the rsync
+        distributor.
+
+        :return: predistributor that was configured in rsync distributor's config
+        :rtype: Distributor
+        """
+        return None
+
+    def _get_postdistributor(self):
+        """
+        Returns the distributor object representing the postdistributor. A postdistirbutor is the
+        distributor used to publish the repository after an rsync publish occurs.
+
+        :return: postdistributor that was configured in rsync distributor's config
+        :rtype: Distributor
+        """
+        postdistributor_id = self.get_config().flatten().get("postdistributor_id", None)
+        return Distributor.objects.get_or_404(repo_id=self.repo.id,
+                                              distributor_id=postdistributor_id)
+
+    def _add_necesary_steps(self, date_filter=None, relative_symlinks=True, config=None):
```

Comment:
`relative_symlinks` needs to be added to the docstring with a `:param:` and `:type:`